docs/tutorials/chatbot.md

```diff
@@ -0,0 +1,728 @@
+    """
+    Download data.
+    :param targets: target path for data.
+    :param download_proxy: download proxy (e.g. 'http', 'https')
+    :param task_name: name of the task
+    """
+    opener = urllib.request.build_opener()
+    opener.addheaders = [('User-agent', 'Mozilla/5.0')]
+    if download_proxy:
+        proxy = urllib.request.ProxyHandler(
+            {'http': download_proxy, 'https': download_proxy}
+        )
+        opener.add_handler(proxy)
+    urllib.request.install_opener(opener)
+    with ProgressBar(description=task_name) as t:
+        for key, value in targets.items():
+            if not os.path.exists(value['filename']):
+                urllib.request.urlretrieve(
+                    value['url'], value['filename'], reporthook=lambda *x: t.update(0.01)
+                )
+def tutorial(args):
+    Path(args.workdir).mkdir(parents=True, exist_ok=True)
+    '''
+    Comment this to use the exectors you have in `executors.py`
+    class MyTransformer(Executor):
+        def foo(self, **kwargs):
+            print(f'foo is doing cool stuff: {kwargs}')
+    
+    class MyIndexer(Executor):
+        def bar(self, **kwargs):
+            print(f'bar is doing cool stuff: {kwargs}')
+    '''
+    targets = {
+        'covid-csv': {
+            'url': args.index_data_url,
+            'filename': os.path.join(args.workdir, 'dataset.csv'),
+        }
+    }
+    # download the data
+    download_data(targets, args.download_proxy, task_name='download covid-dataset')
+    flow = (
+        Flow(cors=True)
+            .add(name='MyTransformer', uses=MyTransformer)
+            .add(name='MyIndexer', uses=MyIndexer)
+            .plot('test.svg')
+    )
+    with flow, open(targets['covid-csv']['filename']) as fp:
+        flow.index(from_csv(fp, field_resolver={'question': 'text'}))
+        
+        # switch to REST gateway at runtime
+        flow.protocol = 'http'
+        flow.port_expose = args.port_expose
+        url_html_path = 'file://' + os.path.abspath(
+            os.path.join(
+                os.path.dirname(os.path.realpath(__file__)), 'static/index.html'
+            )
+        )
+        try:
+            webbrowser.open(url_html_path, new=2)
+        except:
+            pass  # intentional pass, browser support isn't cross-platform
+        finally:
+            default_logger.success(
+                f'You should see a demo page opened in your browser, '
+                f'if not, you may open {url_html_path} manually'
+            )
+        if not args.unblock_query_flow:
+            flow.block()
+if __name__ == '__main__':
+    args = set_hw_chatbot_parser().parse_args()
+    tutorial(args)
+```
+
+And your directory should be:
+
+    .
+    ├── project                    
+    │   ├── app.py          
+    │   ├── my_executors.py         
+    │   └── static/         
+    │   └── our_flow.svg #This will be here if you used the .plot() function       
+    └── ...
```

Comment:
Remove - if the `project` is actually the working dir, just show that

Reply:
no, project is not the workdir. The workdir has a random name, that's why I think the author of the tutorial showed it like that. I'll write random_id/dataset.csv

Reply:
Actually I see that
> We can start by creating an empty folder, I'll call mine `tutorial` and that's the name you'll see through the tutorial but feel free to use whatever you wish.

it's not even called `tutorial`. And I see no reason why the working dir should be outside of it - so let's change this so that we are working inside `tutorial`

Reply:
Removed and replaced by just `dataset.csv`